### DISCUSSION TITLE
`forward_auth` / `reverse_proxy` php_server as upstream

### DISCUSSION THREAD

**Discussion by athopen** [php/frankenphp#1052] [Q&A]
Hey 👋🏻 

I am developing an application where each request to `/` needs to be validated against a PHP backend before serving the Node.js PWA. My current approach is to implement a `reverse_proxy` directive that communicates with the PHP backend, returning a `200` status code on success or redirecting to a different URL in case of failure.

```
{$SERVER_NAME:localhost} {
   ....

    reverse_proxy / localhost:80 {
	rewrite /verify

	@good_response status 2xx
	handle_response @good_response
    }

    @pwa expression `....`

    reverse_proxy @pwa http://{$PWA_UPSTREAM}
   
    php_server
}
```

With this implementation, Caddy would "call itself" for each request to `/`. Isn't there a more efficient way to handle this?

I considered using something like `reverse_proxy / php_server`, but it doesn't seem to work.

**Comment by AlliBalliBaba** [comment]
I don't know if I fully understood what you're trying to achieve, but couldn't you do something like this instead?
```
{$SERVER_NAME:localhost} {
    ....

    handle / {
        rewrite /verify
        php_server
    }
 
    ....
}
```
